cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/wizard/common/control/CheckStateListener.java

```diff
@@ -245,32 +247,52 @@ private void checkPageComplete() {
      * @param packageSelector
      *            current {@link CheckboxTreeViewer} for the package selection
      */
-    private void performCheckLogic(CheckStateChangedEvent event, CheckboxTreeViewer packageSelector) {
+    public void performCheckLogic(CheckStateChangedEvent event, CheckboxTreeViewer packageSelector) {
 
-        SelectIncrementContentProvider cp = (SelectIncrementContentProvider) packageSelector.getContentProvider();
-        TreePath[] paths = cp.getAllPaths(event.getElement());
-        for (TreePath path : paths) {
-            packageSelector.setSubtreeChecked(path, event.getChecked());
-        }
+        if (event.getSource().equals(packageSelector)) {
+            SelectIncrementContentProvider cp = (SelectIncrementContentProvider) packageSelector.getContentProvider();
+            TreePath[] paths = cp.getAllPaths(event.getElement());
+            for (TreePath path : paths) {
+                packageSelector.setSubtreeChecked(path, event.getChecked());
+            }
+
+            TreePath[] parents = cp.getParents(event.getElement());
+            if (event.getChecked()) {
+                for (TreePath parent : parents) {
+                    boolean allChecked = true;
+                    for (Object child : cp.getChildren(parent)) {
+                        if (!packageSelector.getChecked(parent.createChildPath(child))) {
+                            allChecked = false;
+                            break;
+                        }
+                    }
+                    if (allChecked) {
+                        packageSelector.setChecked(parent, true);
+                    }
+                }
 
-        TreePath[] parents = cp.getParents(event.getElement());
-        if (event.getChecked()) {
-            for (TreePath parent : parents) {
-                boolean allChecked = true;
-                for (Object child : cp.getChildren(parent)) {
-                    if (!packageSelector.getChecked(parent.createChildPath(child))) {
-                        allChecked = false;
+                if (event.getElement().toString().contains("All")) {
+                    packageSelector.setAllChecked(true);
+                }
+
+                // checks if all child increments are checked and checks All-Checkbox
+                boolean allChecked2 = true;
```

Comment:
I wouldn't use variable name with numbers. I have never seen them. Please find a better name, or even use `allChecked`. The boolean from above does not conflict with this one